frigate/events/cleanup.py

```diff
@@ -197,9 +199,17 @@ def run(self) -> None:
             self.expire(EventCleanupType.snapshots)
 
             # drop events from db where has_clip and has_snapshot are false
-            delete_query = Event.delete().where(
-                Event.has_clip == False, Event.has_snapshot == False
+            events = (
+                Event.select()
+                .where(Event.has_clip == False, Event.has_snapshot == False)
+                .iterator()
             )
-            delete_query.execute()
+            events_to_delete = [e.id for e in events]
+            if len(events_to_delete) > 0:
+                Event.delete().where(Event.id << events_to_delete).execute()
```

Comment:
This could fail if the number of events is too large. Not sure if its needed here, but we have done this in batches in other places.

Reply:
I can also change this back to do the old where clause delete still, but I need to get the event ids that will be deleted so I can also remove them from the embeddings DB.

Reply:
I actually like breaking it out like this. The sqlite queue does much better with a series of smaller writes so it has time to slot in write requests from other places. Writes that take a long time block that queue, so a series of smaller batches is better.